storage/remote/queue_manager.go

```diff
@@ -545,11 +540,17 @@ func (t *QueueManager) newShards() *shards {
 	return s
 }
 
+type sample struct {
```

Comment:
Hmmm, would be nice if we could use a different name. To me sample implies just ts and value, but I can't think of anything better right now.

Reply:
I could call it something like `queueSample` to differentiate it a bit, but I don't know if that gains anything.